packages/zero-solid/src/solid-view.ts

```diff
@@ -47,60 +49,55 @@ export class SolidView<V> implements Output {
     onTransactionCommit(this.#onTransactionCommit);
     this.#format = format;
     this.#onDestroy = onDestroy;
-    [this.#state, this.#setState] = createStore<State>([
-      {'': format.singular ? undefined : []},
-      queryComplete === true ? complete : unknown,
-    ]);
+    this.#root = {'': format.singular ? undefined : []};
     input.setOutput(this);
 
-    this.#applyChanges(input.fetch({}), node => ({type: 'add', node}));
+    for (const node of input.fetch({})) {
+      applyChange(
+        this.#root,
+        {
+          type: 'add',
+          node,
+        },
+        this.#input.getSchema(),
+        '',
+        this.#format,
+      );
+    }
+
+    [this.#rootStore, this.#setRootStore] = createStore<Entry>(
+      deepClone(this.#root) as Entry,
+    );
+    const [resultDetails, setResultDetails] = createSignal<QueryResultDetails>(
+      queryComplete === true ? complete : unknown,
+    );
+    this.#resultDetails = resultDetails;
 
     if (queryComplete !== true) {
       void queryComplete.then(() => {
-        this.#setState(oldState => [oldState[0], complete]);
+        setResultDetails(complete);
       });
     }
   }
 
   get data(): V {
-    return this.#state[0][''] as V;
+    return this.#rootStore[''] as V;
   }
 
   get resultDetails(): QueryResultDetails {
-    return this.#state[1];
+    return this.#resultDetails();
   }
 
   destroy(): void {
     this.#onDestroy();
   }
 
   #onTransactionCommit = () => {
-    this.#applyChanges(this.#pendingChanges, c => c);
+    this.#setRootStore(reconcile(this.#root));
   };
 
-  #applyChanges<T>(changes: Iterable<T>, mapper: (v: T) => Change): void {
-    try {
-      this.#setState(
-        produce((draftState: State) => {
-          for (const change of changes) {
-            applyChange(
-              draftState[0],
-              mapper(change),
-              this.#input.getSchema(),
-              '',
-              this.#format,
-            );
-          }
-        }),
-      );
-    } finally {
-      this.#pendingChanges = [];
-    }
-  }
-
   push(change: Change): void {
-    // Delay setting the state until the transaction commit.
-    this.#pendingChanges.push(change);
+    applyChange(this.#root, change, this.#input.getSchema(), '', this.#format);
```

Comment:
Is this correct? We are now changing the root without triggering updates. I'm concerned that there might be some way to see this intermediary state?

Maybe we should have a #newRoot and swap the root to that using reconcile in commit?